test/axes/timeAxisTests.ts

```diff
@@ -130,23 +130,62 @@ describe("TimeAxis", () => {
     svg.remove();
   });
 
-  it("tick labels do not overlap with tick marks", () => {
-    let svg = TestMethods.generateSVG(400, 100);
-    scale = new Plottable.Scales.Time();
-    scale.domain([new Date("2009-12-20"), new Date("2011-01-01")]);
-    axis = new Plottable.Axes.Time(scale, "bottom");
-    axis.renderTo(svg);
-    let tickRects = d3.selectAll("." + Plottable.Axis.TICK_MARK_CLASS)[0].map((mark: Element) => mark.getBoundingClientRect());
-    let labelRects = d3.selectAll("." + Plottable.Axis.TICK_LABEL_CLASS)
-        .filter(function(d: Element, i: number) {
-          return d3.select(this).style("visibility") === "visible";
-        })[0].map((label: Element) => label.getBoundingClientRect());
-    labelRects.forEach(function(labelRect: ClientRect) {
-      tickRects.forEach(function(tickRect: ClientRect) {
-        assert.isFalse(Plottable.Utils.DOM.clientRectsOverlap(labelRect, tickRect), "visible label does not overlap with a tick");
+  describe("tick labels and tick marks do not overlap", () => {
+    it("tick labels do not overlap with tick marks in bottom orientation", () => {
+      let svg = TestMethods.generateSVG(400, 100);
+      scale = new Plottable.Scales.Time();
+      scale.domain([new Date("2009-12-20"), new Date("2011-01-01")]);
+      axis = new Plottable.Axes.Time(scale, "bottom");
+      axis.renderTo(svg);
+      let tickRects = d3.selectAll("." + Plottable.Axis.TICK_MARK_CLASS)[0].map((mark: Element) => mark.getBoundingClientRect());
+      let labelRects = d3.selectAll("." + Plottable.Axis.TICK_LABEL_CLASS)
+          .filter(function(d, i) {
+            return d3.select(this).style("visibility") === "visible";
+          })[0].map((label: Element) => label.getBoundingClientRect());
+      labelRects.forEach(function(labelRect: ClientRect) {
+        tickRects.forEach(function(tickRect: ClientRect) {
+          assert.isFalse(Plottable.Utils.DOM.clientRectsOverlap(labelRect, tickRect), "visible label does not overlap with a tick");
+        });
       });
+      svg.remove();
+    });
+
+    function TickMarkAndLabelOverlap() {
+      let tickMarks = d3.selectAll(`.${Plottable.Axis.TICK_MARK_CLASS}:not(.${Plottable.Axis.END_TICK_MARK_CLASS})`);
+      assert.operator(tickMarks.size(), ">=", 1, "There is at least one tick mark in the test");
+      let tickLabels = d3.selectAll(`.${Plottable.Axis.TICK_LABEL_CLASS}`).filter(function(d, i) {
+          return window.getComputedStyle(this).visibility !== "hidden";
+      });
+      assert.operator(tickLabels.size(), ">=", 1, "There is at least one tick label in the test");
+      tickMarks.each(function(d, i) {
+        let tickMarkRect = this.getBoundingClientRect();
+        tickLabels.each(function(e, j) {
```

Comment:
Let's rename the above variables  It is incredibly difficult to understand each piece without a good name